app/support/end_reservation_only.rb

```diff
@@ -0,0 +1,34 @@
+class EndReservationOnly
+  def perform
+    order_details.each do |od|
+      od.transaction do
+        expire_reservation(od)
+      end
+    end
+  end
+
+  private
+
+  def order_details
+    reservation_only_order_details
+  end
+
+  def reservation_only_order_details
+    OrderDetail.purchased_active_reservations
+      .merge(Reservation.in_progress)
```

Comment:
For reservation-only, neither `actual_start_at` nor `actual_end_at` get set. For reservation-only instruments, it is the `reserve_start_at` and `reserve_end_at` that get used for pricing.